utils/dwarf.c

```diff
@@ -261,9 +261,10 @@ static int setup_dwarf_info(const char *filename, struct debug_info *dinfo,
 			    unsigned long offset, bool force)
 {
 	int fd;
+	enum uftrace_trace_type trace_type = check_trace_functions(filename);
 
-	if (!force && check_trace_functions(filename) != TRACE_MCOUNT)
-		return 0;
+	if (force || trace_type == TRACE_MCOUNT || trace_type == TRACE_FENTRY)
+		dinfo->needs_args = true;
```

Comment:
Now I think that `force` is a really bad name.  Anyway we might add some more trace types and miss to update this.  How about this?

```c
if (force || trace_type != TRACE_CYGPROF)
    dinfo->needs_args = true;
```

Reply:
I just updated it as you suggested. Thanks!

Reply:
Why not just moving the check call to the if statement as it was.